sdk/src/main/java/com/mapbox/maps/ResourceOptionsManager.kt

```diff
@@ -20,23 +20,41 @@ class ResourceOptionsManager(
     resourceOptions = resourceOptions.toBuilder().apply(block).build()
   }
 
+  /**
+   * Reset all the properties of the inside resourceOptions to default
+   *
+   * @param context the application context
+   * @param defaultToken the token will applied as default during reset
+   */
+  fun reset(context: Context, defaultToken: String? = null) {
+    resourceOptions = ResourceOptions.Builder().applyDefaultParams(context, defaultToken).build()
```

Comment:
I'm thinking we should use previously set accesstoken as a fallback, instead of throwing runtimeException?

Reply:
👍 since the reset logic is pretty hidden(rarely used), just thinking it might be a surprise it will throw an exception here.